### PR TITLE
Improve code coverage of more Property types in DocumentTraverserTest

### PR DESCRIPTION
**Pull request by BrettMichaelJohnson**

### CODE REVIEW

File: test/com/google/enterprise/adaptor/filenet/ObjectMocks.java [hunk 1]
```diff
@@ -180,6 +201,81 @@ public static void mockDocumentNotFound(MockObjectStore objectStore,
...
+    } else if (PropertyInteger32.class.equals(clazz)) {
+      expect(property.getInteger32Value()).andStubReturn((Integer) value);
+    } else if (PropertyInteger32List.class.equals(clazz)) {
+      expect(property.getInteger32ListValue())
+          .andStubReturn((Integer32List) value);
+    } else if (PropertyString.class.equals(clazz)) {
+      expect(property.getStringValue()).andStubReturn((String) value);
+    } else if (PropertyStringList.class.equals(clazz)) {
+      expect(property.getStringListValue()).andStubReturn((StringList) value);
+    } // Intentionally allow unsupported PropertyTypes to be mocked.
```
Comment (wiarlawd) [inline test/com/google/enterprise/adaptor/filenet/ObjectMocks.java]: By this you just mean there's no else { throw } here?

File: test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java [hunk 2]
```diff
@@ -408,6 +433,156 @@ public void testGetDocContent() throws Exception {
...
   @Test
+  public void testGetDocContent_includedMetadata() throws Exception {
+    options = TestObjectFactory.newConfigOptions(
+        ImmutableMap.of("filenet.includedMetadata", "foo, bar"));
+
+    String id = "{AAAAAAAA-0000-0000-0000-000000000000}";
+    DocId docId = newDocId(new Id(id));
+    MockObjectStore os = getObjectStore();
+    mockDocument(os, id, DOCUMENT_TIMESTAMP, RELEASED, 42d, "text/plain",
+        ImmutableList.<Property>of(
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java]: Done.

File: test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java [hunk 3]
```diff
@@ -408,6 +433,156 @@ public void testGetDocContent() throws Exception {
...
+
+    DocumentTraverser traverser = new DocumentTraverser(options);
+    Request request = new MockRequest(docId);
+    RecordingResponse response = new RecordingResponse();
+    traverser.getDocContent(new Id(id), request, response);
+
+    assertEquals(
+        new Metadata(
+            ImmutableMap.of("foo", "fooValue", "bar", "barValue")
+            .entrySet()),
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java]: It was mostly copy/paste consistency, but I already broke that consistency in another test.

File: test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java [hunk 4]
```diff
@@ -408,6 +433,156 @@ public void testGetDocContent() throws Exception {
...
+        ImmutableMap.of("filenet.includedMetadata", "foo, bar",
+                        "filenet.excludedMetadata", "bar"));
+
+    String id = "{AAAAAAAA-0000-0000-0000-000000000000}";
+    DocId docId = newDocId(new Id(id));
+    MockObjectStore os = getObjectStore();
+    mockDocument(os, id, DOCUMENT_TIMESTAMP, RELEASED, 42d, "text/plain",
+        ImmutableList.<Property>of(
+            mockProperty("foo", "fooValue"),
+            mockProperty("bar", "barValue")),
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java]: Done.

File: test/com/google/enterprise/adaptor/filenet/ObjectMocks.java [hunk 5]
```diff
@@ -180,6 +201,81 @@ public static void mockDocumentNotFound(MockObjectStore objectStore,
...
+    } else {
+      throw new IllegalArgumentException("Unsupported Property type: "
+          + value.getClass().getName());
+    }
+  }
+
+  /**
+   * Returns a Property with the specified type, name, and value.
+   */
+  public static Property mockProperty(Class<? extends Property>  clazz,
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/ObjectMocks.java]: Done.

File: test/com/google/enterprise/adaptor/filenet/ObjectMocks.java (hunk 1 above)
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/ObjectMocks.java]: Yes.  I originally threw IllegalArgumentException (like above), but I needed to test an unsupported type.

File: test/com/google/enterprise/adaptor/filenet/EngineCollectionMocks.java [hunk 6]
```diff
@@ -160,4 +168,52 @@ public int size() {
       return values.size();
     }
   }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/filenet/EngineCollectionMocks.java]: Done.

**Pull request merged**